### ISSUE TITLE
Connector Builder Landing Page

### ISSUE THREAD

**Issue by lmossman** [airbytehq/airbyte#21490]
## Context

Implement the landing page for the connector builder, as shown in this design: https://www.figma.com/proto/F1ljvrJerLoVbNvtejBWid/LOW-CODE-CONNECTOR-BUILDER?page-id=3404%3A119487&node-id=3426%3A123290&viewport=-6573%2C-494%2C2&scaling=min-zoom&starting-point-node-id=3426%3A123290

**Note**: For now, the middle tile (Load an existing connector) and its corresponding page will not be added. This will be added once the "managing multiple connectors" project is complete.

## Acceptance Criteria

A landing page for the connector builder exists with two tiles:
- Import a YAML manifest, which opens the builder in the YAML view containing the imported yaml
- Start from scratch, which for now will just open a blank connector builder

Open question: should we always show this page when going to /connector-builder right now? Or should it only be shown if they don't have anything in local storage?
- Grooming notes: Go with the latter option here to get us incrementally there

**Comment by lmossman** [comment]
Closing in favor of https://github.com/airbytehq/airbyte/issues/21288 and https://github.com/airbytehq/airbyte/issues/21190

**Comment by lmossman** [comment]
Had one round of review and addressed the feedback, waiting on final review
